frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts

```diff
@@ -0,0 +1,334 @@
+
+      // Start migration without waiting for it to finish
+      await vm.action('Migrate', false);
+      await vm.waitForStatusIcon(statusIcons.migrating, VM_MIGRATION_TIMEOUT_SECS);
+
+      await vm.action('Cancel', false);
+      await vm.waitForStatusIcon(statusIcons.running, VM_BOOTUP_TIMEOUT_SECS);
+      await browser.wait(
+        waitForStringInElement(vmDetailNode(vm.namespace, vm.name), sourceNode),
+        VM_MIGRATION_TIMEOUT_SECS,
+      );
+    },
+    VM_ACTIONS_TIMEOUT_SECS,
+  );
+});
+
+describe('Add/remove disks and NICs on respective VM pages', () => {
+  const testVm = getVmManifest('Container', testName, `vm-disk-nic-${testName}`);
+  const vm = new VirtualMachine(testVm.metadata);
+
+  beforeAll(async () => {
+    createResources([multusNad, testVm]);
+    await vm.action('Start');
+  }, VM_BOOTUP_TIMEOUT_SECS);
+
+  afterAll(() => {
+    deleteResources([multusNad, testVm]);
+  });
+
+  it(
+    'Add/remove disk on VM disks page',
+    async () => {
+      await vm.addDisk(hddDisk);
+      expect(await vm.getAttachedDisks()).toContain(hddDisk);
+
+      let vmi = await vm.navigateToVmi(TABS.OVERVIEW);
+      expect((await vmi.getVolumes()).includes(hddDisk.name)).toBe(false);
+
+      await vm.action('Restart');
+
+      vmi = await vm.navigateToVmi(TABS.OVERVIEW);
+      expect((await vmi.getVolumes()).includes(hddDisk.name)).toBe(true);
+
+      await vm.removeDisk(hddDisk.name);
+      expect(await vm.getAttachedDisks()).not.toContain(hddDisk);
+
+      await vm.action('Restart');
+
+      vmi = await vm.navigateToVmi(TABS.OVERVIEW);
+      expect((await vmi.getVolumes()).includes(hddDisk.name)).toBe(false);
+    },
+    VM_ACTIONS_TIMEOUT_SECS * 2,
+  );
+
+  xit(
+    'BZ(1732598) Add/remove nic on VM Network Interfaces page',
+    async () => {
+      await vm.addNIC(networkInterface);
+
+      expect(searchYAML(networkInterface.networkDefinition, vm.name, vm.namespace, 'vmi')).toBe(
+        false,
+      );
+
+      await vm.action('Restart');
+      expect(searchYAML(networkInterface.networkDefinition, vm.name, vm.namespace, 'vmi')).toBe(
+        true,
+      );
+
+      await vm.removeNIC(networkInterface.name);
+      expect((await vm.getAttachedNICs()).includes(networkInterface)).toBe(false);
+
+      await vm.action('Restart');
+
+      expect(searchYAML(networkInterface.networkDefinition, vm.name, vm.namespace, 'vmi')).toBe(
+        false,
+      );
+    },
+    VM_ACTIONS_TIMEOUT_SECS * 2,
+  );
+
+  it('NIC cannot be added twice using one net-attach-def', async () => {
+    await vm.navigateToTab(TABS.NICS);
+    if (
+      (await vm.getAttachedNICs()).filter((nic) => nic.name === networkInterface.name).length === 0
+    ) {
+      await vm.addNIC(networkInterface);
+    }
+
+    // Verify the NIC is added in VM Manifest
+    const resource = getResourceObject(vm.name, vm.namespace, vm.kind);
+    const nic = _.find(
+      _.get(resource, 'spec.template.spec.domain.devices.interfaces', []),
```

Comment:
I believe we have a selector for this. If not, can we add?

Reply:
Could you point to the selectors? I don't know what you mean

Reply:
https://github.com/openshift/console/tree/master/frontend/packages/kubevirt-plugin/src/selectors

In particular, this one seems to be relevant: https://github.com/openshift/console/blob/c0c95dddfc4bce1ba7a35275814ec9281d473e23/frontend/packages/kubevirt-plugin/src/selectors/vm/selectors.ts#L22

Reply:
Either that or `console-shared` selectors, might check out that one as well.